src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        size = 0;
+        head = null;
+        tail = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (size == 0) {
+            head = node;
+            tail = node;
+        } else {
+            node.prev = tail;
+            tail = node;
+            node.prev.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkSizeAdd(index);
+        if (index == 0 && size == 0 || index > 0 && index == size) {
+            add(value);
+        } else {
+            link(value, index);
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return find(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = find(index).value;
+        find(index).value = value;
```

Comment:
You call the function twice, which means you do double work. Get the result into a separate variable, and then work with it.